spec/stallion.rb

```diff
@@ -16,7 +17,7 @@ def ride
     end
 
     def match?(request)
-      method = request['REQUEST_METHOD']
+      method = request.respond_to?(:params) ? request.params['REQUEST_METHOD'] : request['REQUEST_METHOD']
```

Comment:
Reading closer through webrick & rack, I think executing the #params method is the 8 failures requiring Content-Type, because that then goes through union of URL query parameters and POST body as form and joins them in params, which reads body, which makes the body no longer readable.

Calling params, simply put, results in following sequence, that reads the body as a side-effect:
https://github.com/rack/rack/blob/8116397b447e6dc8117c8409cc083bf5b0e2f8fb/lib/rack/request.rb#L553

https://github.com/rack/rack/blob/8116397b447e6dc8117c8409cc083bf5b0e2f8fb/lib/rack/request.rb#L546

https://github.com/rack/rack/blob/8116397b447e6dc8117c8409cc083bf5b0e2f8fb/lib/rack/request.rb#L512

Not sure how earlier rack deals with this, but in Rack >= 3 at least, request has env, which has the REQUEST_METHOD key

IOW I suggest:
```suggestion
      method = request.env['REQUEST_METHOD']
```
which should mean we don't have to touch those tests at all to fix them.

Reply:
Nice, thx for the tip 👍 I have tested with Rack 2.x and this approach seems to work.

Reply:
I have update the PR (could have applied the suggestion, but that have not come to me mind earlier, sorry)